docs/RFCS/20230731_region_liveness.md

```diff
@@ -0,0 +1,363 @@
+   ensure the region is still down. 
+2. If `IsAlive` fails to read from a region, is will trigger a liveness probe
+   for the region.
+
+## Descriptor Leasing Details
+
+### Lease Expiration vs `session_id` Expiration
+
+There are two timestamps that are relevant for leases and lease expiration.
+
+1. The `system.lease.expiration` column.
+2. The `system.sqlliveness.expiration` column. The relationship between a lease
+   and the sqlliveness session is indirect. The `system.lease.nodeID` column
+   stores a `system.sql_instances.id`, which is leased by a
+   `system.sqlliveness.session_id`.
+
+The `system.lease.expiration` timestamp will usually exceed the
+`system.sqlliveness.expiration` time. But the lease subsystem is programmed
+such that a server only assumes the lease lives until the SQL liveness
+session's expiration timestamp. This means it is safe to ignore leases in an
+unavailable region, because creating a `system.sqlliveness.session_id` to claim
+the `sql_instance.id` row that owns the unexpired lease will require marking
+the region as available, which will delete all stale leases in the region
+before servers are allowed to start up.
+
+### Pseudo Code for Advancing Descriptor Version
+
+A descriptor version is allowed to advance to version n+1 as long as there are
+no outstanding leases for version n-1. When searching for descriptor leases,
+the lease subsystem will skip unavailable regions.
+
+```
+txn := BeginTxn()
+regionLiveness := ReadRegionLiveness(txn)
+for region in AllRegions():
+    unavailableAt, exists := regionLiveness[region]
+    if exists and unavailableAt.Before(txn.Time()):
+       continue // skip unavailable regions
+    if hasLease(region, descriptor, version):
+       return true
+return false
+```
+
+## SQL Instances
+
+### ID Allocation
+
+The `system.sql_instances.id` allocation loop will be rewritten to ignore
+unavailable regions. This is safe because instances in the region will be
+deleted by the transaction that restores region availability.
+
+### Instance Range Feed
+
+There is a rangefeed over the `system.sql_instances` table. The rangfeed
+maintains an instance cache that is used to find servers for distsql. There are
+two changes required to the cache that maintains the rangefeed.
+1. The cache might see duplicate instance ids if a region is unavailable and an
+   instance id used by the unavailable region is claimed by an available
+   region. An easy way to work around this is to store the
+   `system.sql_instances` row in the cache with the more recent transaction
+   timestamp.
+2. The rangfeed should be split by region so that the cache can be initialized
+   when a region is unavailable.
+
+## Jobs
+
+The job subsystem runs a periodic scan to reclaim jobs acquired by a dead
+`system.sqlliveness.session_id`. The Jobs claim loop uses the
+`sqlliveness.Reader.IsAlive`, which will be enhanced to consult
+`system.region_liveness`, so job system specific changes are needed to benefit
```

Comment:
Am I misunderstanding something or is there a missing "no" or "not" from this sentence?

 I'm reading this as saying "The API used by the job system is being upgraded to account for `region_liveness` anyways. Therefore there are [no] job system specific changes needed"